File: patterns/countdown-timer/rules-js/countdown-timer.js
rules.JSRule({
  name: "Countdown-Verwaltung",
  triggers: [triggers.ItemCommandTrigger("myCounter")],
  execute: (event) => {
    const cmmd = parseInt(event.receivedCommand);
    let count = 0;
    const state = items.getItem("myCounter").rawState;
    if (state != null && state.toString() !== "NULL") {
      count = parseInt(items.getItem("myCounter").state);
    }

    if (cmmd === -1 && count > 0) {
      if (count === 1) {
        items.getItem("testLamp").sendCommand("OFF");
      }
      items.getItem("myCounter").postUpdate(count - 1);
    } else if (cmmd >= count || cmmd < -1) {
      let newCount = cmmd < -1 ? -cmmd : cmmd;
      items.getItem("myCounter").postUpdate(newCount);
      if (items.getItem("testLamp").state !== "ON") {
        items.getItem("testLamp").sendCommand("ON");
      }
    } else if (cmmd === 0) {
      items.getItem("myCounter").postUpdate(0);
      items.getItem("testLamp").sendCommand("OFF");
    }
  }
});

rules.JSRule({
  name: "6 Minuten starten",
  triggers: [triggers.ItemCommandTrigger("test6")],
  execute: (event) => { items.getItem("myCounter").sendCommand(6); }
});

rules.JSRule({
  name: "3 Minuten starten",
  triggers: [triggers.ItemCommandTrigger("test3")],
  execute: (event) => { items.getItem("myCounter").sendCommand(3); }
});

rules.JSRule({
  name: "auf 2 Minuten setzen",
  triggers: [triggers.ItemCommandTrigger("test2")],
  execute: (event) => { items.getItem("myCounter").sendCommand(-2); }
});

rules.JSRule({
  name: "Countdown abbrechen",
  triggers: [triggers.ItemCommandTrigger("testabort")],
  execute: (event) => { items.getItem("myCounter").sendCommand(0); }
});

File: patterns/countdown-timer/rules-python/countdown_timer.py
from openhab import rule, Registry
from openhab.triggers import ItemCommandTrigger


@rule(triggers=[ItemCommandTrigger("myCounter")])
class CountdownVerwaltung:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        cmmd = int(str(event.getItemCommand()))
        count = 0
        state = Registry.getItem("myCounter").getState()
        if str(state) != "NULL":
            count = int(str(state))

        if cmmd == -1 and count > 0:
            if count == 1:
                Registry.getItem("testLamp").sendCommand("OFF")
            Registry.getItem("myCounter").postUpdate(count - 1)
        elif cmmd >= count or cmmd < -1:
            new_count = -cmmd if cmmd < -1 else cmmd
            Registry.getItem("myCounter").postUpdate(new_count)
            if str(Registry.getItem("testLamp").getState()) != "ON":
                Registry.getItem("testLamp").sendCommand("ON")
        elif cmmd == 0:
            Registry.getItem("myCounter").postUpdate(0)
            Registry.getItem("testLamp").sendCommand("OFF")


@rule(triggers=[ItemCommandTrigger("test6")])
class SechsMinutenStarten:
    def execute(self, module, input):
        Registry.getItem("myCounter").sendCommand(6)


@rule(triggers=[ItemCommandTrigger("test3")])
class DreiMinutenStarten:
    def execute(self, module, input):
        Registry.getItem("myCounter").sendCommand(3)


@rule(triggers=[ItemCommandTrigger("test2")])
class AufZweiMinutenSetzen:
    def execute(self, module, input):
        Registry.getItem("myCounter").sendCommand(-2)


@rule(triggers=[ItemCommandTrigger("testabort")])
class CountdownAbbrechen:
    def execute(self, module, input):
        Registry.getItem("myCounter").sendCommand(0)
